cpp/include/raft/core/device_container_policy.hpp

```diff
@@ -183,4 +183,43 @@ class device_uvector_policy {
   [[nodiscard]] auto make_accessor_policy() const noexcept { return const_accessor_policy{}; }
 };
 
+/**
+ * @brief A container policy for device mdarray allocated in the temporary workspace.
+ */
+template <typename ElementType>
+class device_uvector_workspace_policy {
```

Comment:
I like the idea of having a different container policy specifically for workspace resources but is there any reason we can't just subclass the current device container policy and override create()?

Reply:
I guess we can! The only problem we would still have to verbosely re-export all of the type aliases, right?

Reply:
I've opted to use only one policy and pass a memory resource to it, as before. The reason is that the policy type is aliased within mdarray type; so the arrays created with different policies would have different types, which would require a too large refactoring and may be difficult to use.